src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateArray = [];
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+
+        break;
+
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const keys of action.keysToRemove) {
+          delete stateCopy[keys];
+        }
```

Comment:
Don't forget to add a default case in case your type doesn't match any case

Reply:
thanks, but you forget to approve) according to the terms of the task, we have only 3 types of case ( add, remove and clear)